pkg/controller/volume/cache/desired_state_of_world.go

```diff
@@ -57,7 +57,7 @@ type DesiredStateOfWorld interface {
 	// should be attached to the specified node, the volume is implicitly added.
 	// If no node with the name nodeName exists in list of nodes managed by the
 	// attach/detach attached controller, an error is returned.
-	AddPod(podName types.UniquePodName, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
+	AddPod(pod *api.Pod, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
```

Comment:
Sorry I was unclear. I'm ok with the pod object being passed in instead of individual pod fields, but please leave `podName types.UniquePodName` as is. I want the caller to be very explicit aware when calling this method what will be used as the key to refer to this pod.